src/bokeh/io/notebook.py

```diff
@@ -498,6 +500,41 @@ def publish_display_data(data: dict[str, Any], metadata: dict[Any, Any] | None =
     from IPython.display import publish_display_data
     publish_display_data(data, metadata, transient=transient, **kwargs)
 
+def remote_jupyter_proxy_url(port: int):
+    """Callable to configure Bokeh's show method when a proxy must be
+    configured. If port is None we're asking about the URL
+    for the origin header.
+
+    Taken from documentation here:
+
+       https://docs.bokeh.org/en/latest/docs/user_guide/output/jupyter.html#jupyterhub
+
+    and made an implicit override when JUPYTER_BOKEH_EXTERNAL_URL is defined in
+    a user's environment to the external hostname of the hub, e.g. https://our-hub.edu
+
+    Args:
+       port (int):
+           random port generated by bokeh to avoid re-using recently closed ports
+
+       Returns:
+           str:
+               URL capable of traversing the JupyterHub proxy to return to this notebook session.
+    """
+    base_url = os.environ['JUPYTER_BOKEH_EXTERNAL_URL']
```

Comment:
Same comment here. I know you have checked for the existence of this before calling this function, but to be honest I think it might be better to not do that, and instead just have a function that encapsulates updating the `notebook_url` from the environment, that encapsulates any and all `environ` access inside it (I don't love that `show_app` is directly looking at `os.environ`) 

Alternatively,  if we want to make all these assumptions, then at least we need to make this function "private" by renaming to `_remote_jupyter_proxy_url`

Reply:
I was debating _remote_jupyter_proxy_url myself and that sounds like the best move to me.   Coming up.